src/Native/System.Native/pal_readdir.h

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+#include <stdint.h>
+#include <dirent.h>
+
+enum class NodeType : int16_t
+{
+    PAL_DT_UNKNOWN  =  0,
+    PAL_DT_FIFO     =  1,
+    PAL_DT_CHR      =  2,
+    PAL_DT_DIR      =  4,
+    PAL_DT_BLK      =  6,
+    PAL_DT_REG      =  8,
+    PAL_DT_LNK      = 10,
+    PAL_DT_SOCK     = 12,
+    PAL_DT_WHT      = 14
+};
```

Comment:
Are these numeric values specified by POSIX?

Add static_asserts that PAL_DT_* ==  DT_* like I did for S_*.